### PR TITLE
Common CERT oracle

### PR DESCRIPTION
**Pull request by malwaregarry**

### CODE REVIEW

**Review by mrigger — APPROVED** [review]

File: src/sqlancer/cockroachdb/CockroachDBOracleFactory.java [hunk 1]
```diff
@@ -97,7 +100,33 @@ public TestOracle<CockroachDBProvider.CockroachDBGlobalState> create(
...
+                        return Optional.of(number);
+                    } catch (Exception e) { // To avoid the situation that no number is found
+                    }
+                }
+                return Optional.empty();
+            };
+            CERTOracle.CheckedFunction<SQLancerResultSet, Optional<String>> queryPlanParser = (rs) -> {
+                String content = rs.getString(1);
+                if (content.contains("• ")) {
+                    String operation = content.split("• ")[1].split(" ")[0];
```
Comment (mrigger) [inline src/sqlancer/cockroachdb/CockroachDBOracleFactory.java:120]: The parsing logic for the query plan seems quite brittle. At some point, I guess we should extract these to a method and then have tests that check whether the logic still works when the database systems are updated.

**Comment by mrigger** [comment]
@bajinsheng, could you perhaps also a quick look whether this looks good to you?

File: src/sqlancer/tidb/TiDBExpressionGenerator.java [hunk 2]
```diff
@@ -60,68 +57,12 @@ private enum Gen {
...
-            TiDBFunction func = TiDBFunction.getRandom();
-            return new TiDBFunctionCall(func, generateExpressions(func.getNrArgs(), depth));
-        case BINARY_BIT:
-            return new TiDBBinaryBitOperation(generateExpression(depth + 1), generateExpression(depth + 1),
-                    TiDBBinaryBitOperator.getRandom());
-        case BINARY_LOGICAL:
-            return new TiDBBinaryLogicalOperation(generateExpression(depth + 1), generateExpression(depth + 1),
-                    TiDBBinaryLogicalOperator.getRandom());
-        case CAST:
-            return new TiDBCastOperation(generateExpression(depth + 1), Randomly.fromOptions("BINARY", // https://github.com/tidb-challenge-program/bug-hunting-issue/issues/52
```
Comment (bajinsheng) [inline src/sqlancer/tidb/TiDBExpressionGenerator.java:109]: These issues have already been fixed, so we can remove them later.

**Review by bajinsheng — COMMENTED** [review]
LGTM!

**Review by bajinsheng — APPROVED** [review]

**Pull request merged**